server-common/src/test/java/org/apache/kafka/server/common/MetadataVersionTest.java

```diff
@@ -466,6 +467,15 @@ public void assertLatestProductionIsLessThanLatest() {
             " to be less than the latest of " + MetadataVersion.latestTesting());
     }
 
+    @Test
+    public void testLastMetadataProductionDontReturnUnstableVersion() {
```

Comment:
Perhaps add a comment like the following.

`The broker picks the version for a few inter broker RPCs based on the metadata version, instead of the supported version from ApiResponse. We need to make sure that the latest production MV doesn't accidentally depend on an unstable request version.`